eng/common/scripts/GeneralReleaseNotesLogic.ps1

```diff
@@ -0,0 +1,129 @@
+. (Join-Path $PSScriptRoot common.ps1)
+
+$PATTERN_REGEX = "^\[pattern(\.(?<SectionName>\w+))?\]:\s#\s\((?<Pattern>.*)\)"
+$CsvMetaData = Get-CSVMetadata
+
+function Get-PackagesInfoFromFile ($releaseNotesLocation) 
+{
+    $releaseNotesContent = Get-Content -Path $releaseNotesLocation
+    $checkLine = $False
+    $presentPkgInfo = @()
+
+    foreach ($line in $releaseNotesContent)
+    {
+        $line = $line.Trim()
+        if ($line.StartsWith("<!--"))
+        {
+            $checkLine = $True
+            continue
+        }
+        if ($line.EndsWith("-->"))
+        {
+            break
+        }
+        if ($checkLine)
+        {
+            $pkgInfo = $line.Split(":")
+            $packageName = $pkgInfo[0]
+            $packageMetaData = $CsvMetaData | Where-Object { $_.Package -eq $packageName }
+            if ($packageMetaData.Count -gt 0)
+            {
+                $presentPkgInfo += $line
+            }
+        }
+    }
+    return $presentPkgInfo
+}
+
+function Filter-ReleaseHighlights ($releaseHighlights)
+{
+    $results = @{}
+
+    foreach ($key in $releaseHighlights.Keys)
+    {
+        $keyInfo = $key.Split(":")
+        $packageName = $keyInfo[0]
+        $packageVersion = $keyInfo[1]
+
+        $packageMetaData = $CsvMetaData | Where-Object { $_.Package -eq $packageName }
+
+        if ($packageMetaData.ServiceName -eq "template")
+        {
+            continue
+        }
+
+        $existingPackages = GetExistingPackageVersions -PackageName $packageName `
+        -GroupId $packageMetaData.GroupId
+
+        $versionExists = $existingPackages | Where-Object { $_ -eq $packageVersion }
+
+        if ($null -eq $versionExists)
+        {
+            continue
+        }
+
+        $results.Add($key, $releaseHighlights[$key])
```

Comment:
Add in the displayname property here instead of re-filtering it from the csv later.

Reply:
Via add-member https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.utility/add-member?view=powershell-7.1